src/Rules/Gmp.hs

```diff
@@ -24,7 +24,7 @@ gmpContext = vanillaContext Stage1 integerGmp
 
 -- | Build directory for in-tree GMP library.
 gmpBuildPath :: Action FilePath
-gmpBuildPath = buildRoot <&> (-/- stageString (stage gmpContext) -/- "gmp")
+gmpBuildPath = buildRoot <&> (-/- buildDir gmpContext -/- "gmp")
```

Comment:
Why `buildDir` and not `contextDir`? Adding an extra `build` layer seems unnecessary.

Reply:
If this still compiles with the `contextDir` sure...

Reply:
I tried changing all `buildPath`/`buildDir`/`getBuildPath` to the corresponding `context*` variant, and there are some tricky problems when I try building ghc. Most notably everything that handles autogen paths and what not. Maybe I'm just tired but @snowleopard if you're not happy with this, could you perhaps raise an issue about trying to get rid of this, but keep it for another PR?